src/modm/board/disco_f411ve/rcc_prototype.hpp

```diff
@@ -0,0 +1,58 @@
+#pragma once
+
+#include <modm/architecture/interface/clock.hpp>
+
+namespace modm::platform
+{
+
+// @todo integrate with /platform/clock/stm32/rcc.hpp.in
+class RccProto
+{
+public:
+	template<Rcc::AhbPrescaler Prescaler>
+	static consteval int
+	prescalerToValue()
+	{
+		switch (Prescaler)
+		{
+			case Rcc::AhbPrescaler::Div1: return 1;
+			case Rcc::AhbPrescaler::Div2: return 2;
+			case Rcc::AhbPrescaler::Div4: return 4;
+			case Rcc::AhbPrescaler::Div8: return 8;
+			case Rcc::AhbPrescaler::Div16: return 16;
+			case Rcc::AhbPrescaler::Div64: return 64;
+			case Rcc::AhbPrescaler::Div128: return 128;
+			case Rcc::AhbPrescaler::Div256: return 256;
+			case Rcc::AhbPrescaler::Div512: return 512;
+		};
+	}
+
+	template<Rcc::Apb1Prescaler Prescaler>
+	static consteval int
+	prescalerToValue()
+	{
+		switch (Prescaler)
+		{
+			case Rcc::Apb1Prescaler::Div1: return 1;
+			case Rcc::Apb1Prescaler::Div2: return 2;
+			case Rcc::Apb1Prescaler::Div4: return 4;
+			case Rcc::Apb1Prescaler::Div8: return 8;
+			case Rcc::Apb1Prescaler::Div16: return 16;
+		};
+	}
```

Comment:
To associate prescaler values to their registers, i've borrowed this construct from modm::platform::GeneralPurposeTimer::signalToChannel<...>(...) but using a switch operator.

Reply:
Sure, but I'd rather use a normal constexpr function like `uint8_t Rcc::Apb1Div(Rcc::Apb1Prescaler pre) {...}`, then it can also be used at runtime if necessary.

Reply:
of course!